adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -119,6 +118,480 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
+      val leftRdd: GenomicRDD[T, U] = if (partitions != rdd.partitions.length) evenlyRepartition(partitions) else self
+      val preparedRdds = prepareForShuffleRegionJoin(leftRdd, genomicRdd)
+      val leftRddReadyToJoin = preparedRdds._1
+      val rightRddReadyToJoin = preparedRdds._2
+      assert(leftRddReadyToJoin.partitions.length == rightRddReadyToJoin.partitions.length)
+
+      // what sequences do we wind up with at the end?
+      val endSequences = sequences ++ genomicRdd.sequences
+
+      GenericGenomicRDD[(T, X)](
+        InnerShuffleRegionJoin[T, X](endSequences, endSequences.records.map(_.length).sum / partitions, rdd.context)
+          .joinCoPartitionedRdds(leftRddReadyToJoin, rightRddReadyToJoin),
+        endSequences,
+        kv => {
+          getReferenceRegions(kv._1) ++ genomicRdd.getReferenceRegions(kv._2)
+        }).asInstanceOf[GenomicRDD[(T, X), Z]]
+    }
+
+    /**
+     * Performs a sort-merge inner join between this RDD and another RDD,
+     * followed by a groupBy on the left value.
+     *
+     * In a sort-merge join, both RDDs are co-partitioned and sorted. The
+     * partitions are then zipped, and we do a merge join on each partition.
+     * The key equality function used for this join is the reference region
+     * overlap function. Since this is an inner join, all values who do not
+     * overlap a value from the other RDD are dropped. In the same operation,
+     * we group all values by the left item in the RDD.
+     *
+     * @param genomicRdd The right RDD in the join.
+     * @return Returns a new genomic RDD containing all pairs of keys that
+     *   overlapped in the genomic coordinate space, grouped together by
+     *   the value they overlapped in the left RDD..
+     */
+    def shuffleRegionJoinAndGroupByLeft[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(T, Iterable[X]), Z]](genomicRdd: GenomicRDD[X, Y],
+                                                                                                        optPartitions: Option[Int] = None)(implicit tTag: ClassTag[T], xTag: ClassTag[X]): GenomicRDD[(T, Iterable[X]), Z] = {
+      // did the user provide a set partition count?
+      // if no, we will avoid repartitioning this one
+      val partitions = optPartitions.getOrElse(rdd.partitions.length)
+      val leftRdd: GenomicRDD[T, U] = if (partitions != rdd.partitions.length) evenlyRepartition(partitions) else self
+      val preparedRdds = prepareForShuffleRegionJoin(leftRdd, genomicRdd)
+      val leftRddReadyToJoin = preparedRdds._1
+      val rightRddReadyToJoin = preparedRdds._2
+      assert(leftRddReadyToJoin.partitions.length == rightRddReadyToJoin.partitions.length)
+
+      // what sequences do we wind up with at the end?
+      val endSequences = sequences ++ genomicRdd.sequences
+
+      assert(leftRddReadyToJoin.partitions.length == rightRddReadyToJoin.partitions.length)
+
+      val joinedRDD = InnerShuffleRegionJoin[T, X](endSequences,
```

Comment:
This shouldn't be changed. This should be using `InnerShuffleRegionJoinAndGroupByLeft`.